tests/suites/test_suite_ecp.function

```diff
@@ -15,6 +15,43 @@
 #define ECP_PT_RESET( x )           \
     mbedtls_ecp_point_free( x );    \
     mbedtls_ecp_point_init( x );
+
+#if defined(MBEDTLS_TEST_HOOKS) && defined(MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED)
+/* Test whether bytes represents (in big-endian base 256) a number B that
+ * is "significantly" above a power of 2, which is defined as follows.
+ * Let n be the integer such that 2^n <= B < 2^{n+1}. B is significantly
+ * above a power of 2 if (B - 2^n) / 2^n is not negligible. "Negligible"
+ * is defined as having a negligible chance that if you draw an integer
+ * in the range [1, B-1] K times, the number will always be less than 2^n,
+ * where K is the iteration count passed to genkey_sw_many.
+ */
+static int is_significantly_above_a_power_of_2( data_t *bytes )
+{
+    const uint8_t *p = bytes->x;
+    size_t len = bytes->len;
+    unsigned x;
+    while( len > 0 && p[0] == 0 )
+    {
+        ++p;
+        --len;
+    }
+    if( len == 0 )
+        return( 0 );
+    else if( len == 1 )
+        x = p[0];
+    else
+        x = ( p[0] << 8 ) | p[1];
+
+    if( x <= 4 )
```

Comment:
I've tried to make the function clear. It's not easy because it relies on heuristics that work for the current test data, but would not necessarily work for arbitrary data.

I'm considering removing this function and requiring the test data to indicate when there are enough iterations so that the statistical checks are unlikely to fail. But this would require having basically the same explanation in a comment, without the benefit of code to avoid having to manually evaluate each test case.

Reply:
I've been thinking perhaps the function could be rewritten in a completely different way to increase readability perhaps at the expense of performance (but you know what they say about premature optimisation): parse the bytes to an MPI and then work on that MPI (for example subtract 2^n where n is it bit length and check if the result has a bitlength much smaller than n). Wdyt?

(Note: posted before I read your new version, just some idea I had yesterday after leaving work.)

Reply:
Yes, you're right, it would probably make the function a easier to understand. There'd still be unjustified heuristics but at least the math would be easier to follow. Let me know if the updated code is good enough or if you want me to switch to MPI operations.